auth-pam.c

```diff
@@ -1077,6 +1091,22 @@ KbdintDevice mm_sshpam_device = {
 	mm_sshpam_free_ctx
 };
 
+KbdintDevice sshpam_device_legacy = {
+	"pam-legacy-instructions",
+	sshpam_legacy_init_ctx,
+	sshpam_query,
+	sshpam_respond,
+	sshpam_free_ctx
+};
+
+KbdintDevice mm_sshpam_device_legacy = {
```

Comment:
Reiterating an earlier comment: I don't think we should add this unless there is a clear motivating case of a client that is broken by these changes. Do you have an example of one?

An alternative to adding a whole other method is to add a `compat.c` pattern to detect these clients and automatically adapt.

Reply:
Yeah, I indeed have no case for this, so we can drop it and if the problem shows for some older clients (most wouldn't work anyways since they don't support other features) then we can reiterate adding compatibility code.

Are you fine with that?

Another reason why this was added was not to change the behavior for some clients who are expecting the old way, mostly because if a client pre-this-PR will connect to a server with this PR, it may not receive valid UTF-8 chars output, and this could be an issue.

By providing an option, it's possible to use the old behavior. But that's the only case I may think we could create problems.